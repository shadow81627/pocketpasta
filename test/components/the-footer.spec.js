import Component from '@/components/Layout/the-footer';
import { shallowMount, createLocalVue, RouterLinkStub } from '@vue/test-utils';
import Vuex from 'vuex';
import Vuetify from 'vuetify';
import Vue from 'vue';

const localVue = createLocalVue();
let vuetify;

localVue.use(Vuex);
Vue.use(Vuetify);

describe('Footer', () => {
  beforeEach(() => {
    vuetify = new Vuetify();
  });

  const factory = () =>
    shallowMount(Component, {
      localVue,
      vuetify,
      stubs: {
        NuxtLink: RouterLinkStub,
      },
      mocks: {
        $config: {
          VERSION: '0.11.19',
          COMMIT: 'a95a529a01ab4fe627cc276b35ed8f5e0c6f45d9',
        },
      },
      data: () => ({
        utc: true,
        lastModified: null,
      }),
    });

  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = factory();
    wrapper.setData({ lastModified: 'Sat 19 Oct 2019 09:11:49 GMT+0000' });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders properly without lastModified', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
