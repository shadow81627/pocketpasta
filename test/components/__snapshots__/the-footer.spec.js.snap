// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders properly 1`] = `
"<v-footer-stub height=\\"auto\\" tag=\\"div\\" tile=\\"true\\" absolute=\\"true\\" app=\\"true\\" inset=\\"true\\" class=\\"hidden-print-only\\">
  <div class=\\"container-fluid\\">
    <v-row-stub tag=\\"div\\" nogutters=\\"true\\">
      <v-col-stub cols=\\"auto\\" tag=\\"div\\">
        <v-card-stub loaderheight=\\"4\\" tag=\\"div\\" color=\\"transparent\\" flat=\\"true\\">
          <v-card-text-stub><span>Version: 0.11.19</span> <span>|</span> <span>Commit: </span> <a><span> a95a529</span></a></v-card-text-stub>
        </v-card-stub>
      </v-col-stub>
      <v-spacer-stub></v-spacer-stub>
      <v-col-stub cols=\\"auto\\" tag=\\"div\\">
        <v-card-stub loaderheight=\\"4\\" tag=\\"div\\" color=\\"transparent\\" flat=\\"true\\">
          <v-card-text-stub><span>This site is protected by reCAPTCHA and the Google</span> <a href=\\"https://policies.google.com/privacy\\">Privacy Policy</a> <span>and</span> <a href=\\"https://policies.google.com/terms\\">Terms of Service</a> <span>apply.</span></v-card-text-stub>
        </v-card-stub>
      </v-col-stub>
    </v-row-stub>
    <v-row-stub tag=\\"div\\" nogutters=\\"true\\">
      <v-col-stub cols=\\"auto\\" tag=\\"div\\">
        <v-card-stub loaderheight=\\"4\\" tag=\\"div\\" color=\\"transparent\\" tile=\\"true\\" flat=\\"true\\">
          <v-card-text-stub class=\\"py-2 text-left\\"><span>Made with </span> <span>üçù</span> <span> in Brisbane, Australia</span> <br> <span>¬© 2015 | PocketPasta</span></v-card-text-stub>
        </v-card-stub>
      </v-col-stub>
      <v-spacer-stub></v-spacer-stub>
      <lastmodified-stub utc=\\"true\\"></lastmodified-stub>
    </v-row-stub>
  </div>
</v-footer-stub>"
`;

exports[`Footer renders properly without lastModified 1`] = `
"<v-footer-stub height=\\"auto\\" tag=\\"div\\" tile=\\"true\\" absolute=\\"true\\" app=\\"true\\" inset=\\"true\\" class=\\"hidden-print-only\\">
  <div class=\\"container-fluid\\">
    <v-row-stub tag=\\"div\\" nogutters=\\"true\\">
      <v-col-stub cols=\\"auto\\" tag=\\"div\\">
        <v-card-stub loaderheight=\\"4\\" tag=\\"div\\" color=\\"transparent\\" flat=\\"true\\">
          <v-card-text-stub><span>Version: 0.11.19</span> <span>|</span> <span>Commit: </span> <a><span> a95a529</span></a></v-card-text-stub>
        </v-card-stub>
      </v-col-stub>
      <v-spacer-stub></v-spacer-stub>
      <v-col-stub cols=\\"auto\\" tag=\\"div\\">
        <v-card-stub loaderheight=\\"4\\" tag=\\"div\\" color=\\"transparent\\" flat=\\"true\\">
          <v-card-text-stub><span>This site is protected by reCAPTCHA and the Google</span> <a href=\\"https://policies.google.com/privacy\\">Privacy Policy</a> <span>and</span> <a href=\\"https://policies.google.com/terms\\">Terms of Service</a> <span>apply.</span></v-card-text-stub>
        </v-card-stub>
      </v-col-stub>
    </v-row-stub>
    <v-row-stub tag=\\"div\\" nogutters=\\"true\\">
      <v-col-stub cols=\\"auto\\" tag=\\"div\\">
        <v-card-stub loaderheight=\\"4\\" tag=\\"div\\" color=\\"transparent\\" tile=\\"true\\" flat=\\"true\\">
          <v-card-text-stub class=\\"py-2 text-left\\"><span>Made with </span> <span>üçù</span> <span> in Brisbane, Australia</span> <br> <span>¬© 2015 | PocketPasta</span></v-card-text-stub>
        </v-card-stub>
      </v-col-stub>
      <v-spacer-stub></v-spacer-stub>
      <lastmodified-stub utc=\\"true\\"></lastmodified-stub>
    </v-row-stub>
  </div>
</v-footer-stub>"
`;
